var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -126,13 +126,20 @@ class Boost(Package):
     depends_on('mpi', when='+mpi')
     depends_on('bzip2', when='+iostreams')
     depends_on('zlib', when='+iostreams')
+    depends_on('zlib', when='%xl_r' )
```

Comment:
I would like to build zlib and bzip2 dependencies with gcc, but was not able to do it without it. 

spack install boost@1.62.0%xl_r^zlib%gcc^bzip2%gcc